epm/Distribution/Client/IndexUtils.hs

```diff
@@ -450,7 +451,7 @@ packageIndexFromCache mkPkg hnd entrs mode = accum mempty [] entrs
       -- from the index tarball if it turns out that we need it.
       -- Most of the time we only need the package id.
       ~(pkg, pkgtxt) <- unsafeInterleaveIO $ do
-        mPatch <- patchedPackageCabalFile pkgid
+        mPatch <- patchedPackageCabalFile pkgid defaultPatchesDir
```

Comment:
I don't think removing and recloning is required. You just need to discard the unstaged changes in git before updating it. I will tinker more with this and file an separate issue. We can probably then decide, if it's worth to implement it. Even doing `cp` will mess up the git repository. I think having a separate interface (i.e a flag for epm update) for this with `epm` is likely a good solution.